### ISSUE TITLE
RagTag and regions of heterozygosity

### ISSUE THREAD

**Issue by agolicz** [malonge/RagTag#104]
Hi,
I have a somewhat hypothetical question and am wondering how would RagTag handle a situation like that. Let's say that my genome is mostly inbred with some residual heterozygosity (95% inbred, 5% heterozygous), so for 5% of the genome I could potentially have contigs which correspond to two haplotypes. Those would likely map to the same location in the reference genome (representing single haplotype). While scaffolding using an existing genome would RagTag be likely to place contigs representing one of the haplotypes and then leave the second haplotype unplaced? I suppose what I am wondering is how are mapping conflicts which could result from existence of two haplotypes (corresponding to single haplotype in the reference) resolved. Sorry if this is not entirely clear...

All the best,
Agnieszka

**Comment by malonge** [comment]
Hi Agnieszka,

This is a good and common question. RagTag expects an assembly that only represents one haplotype at any given locus. Therefore, RagTag would likely place the redundant haplotigs that you describe next to each other in the scaffolds causing a false duplication.

So I definitely recommend either using a phased or properly squashed assembly. For squashed assemblies, I recommend running something like "purge_dups" to remove any alternate contigs.

Let me know if that is clear and if it answers your question.

Thanks,
Mike

**Comment by agolicz** [comment]
Hi Mike,
Many thanks. I will leave some details about our case here if someone stumbles upon this thread in the future and finds this helpful.
We sequenced an inbred plant genome with HiFi at 10x. The k-mer plots from hifiasm and KAT and coverage plot from purge_haplotigs did not have any hetrozygous peaks, but BUSCO results did indicate some duplication (this was not conclusive as some plant species do have higher BUSCO duplication levels).
KAT plot
![plot 27](https://user-images.githubusercontent.com/10213846/155310127-f97d46a8-ea14-439f-b2c9-fe51323b2e88.png)

purge_haplotigs plot
![aligned bam histogram](https://user-images.githubusercontent.com/10213846/155310245-b111c739-1048-4f8c-b5a1-590e2404173b.png)

However, when I was looking at tandem duplicated genes and compared two genotypes from the same species our genotype appeared to have higher number of tandem duplicated genes (hence my question above - what happens to haplotigs during scaffolding), which together with BUSCO results made me wonder about some potential heterozygostiy.
Now instead of plotting k-mers or overall coverage we calculated mean coverage for all contigs and plotted that. As a result we got a bimodal distribution. 

Plot of mean contig coverages
![Density contigs coverage all](https://user-images.githubusercontent.com/10213846/155310395-411dc9d8-f3e7-4ecd-91aa-13b0f53d37d4.png)

The heterozygous peak actually corresponds to a large number of short contigs, which may be why it did not show up on on the overall k-mer/coverage analysis (that's just my guess, it looks like we have rather short regions of heterozygosity interspersed in otherwise largely homozygous diploid genome and we had pretty low sequencing coverage).
We've now run purge_haplotigs (using cutoffs from the contig plot), which seems to have largely resolved the issue. Now onto re-scaffolding with RagTag.

Thanks again. Your reply helped a lot!

All the best,
Agnieszka
